ksqldb-engine/src/main/java/io/confluent/ksql/util/SandboxedSharedKafkaStreamsRuntimeImpl.java

```diff
@@ -71,9 +72,36 @@ public void register(
       final QueryId queryId
   ) {
     collocatedQueries.put(queryId, binpackedPersistentQueryMetadata);
-    if (!kafkaStreams.getTopologyByName(
-        binpackedPersistentQueryMetadata.getQueryId().toString()).isPresent()) {
-      kafkaStreams.addNamedTopology(binpackedPersistentQueryMetadata.getTopologyCopy(this));
+    try {
+      if (!kafkaStreams.getTopologyByName(
+          binpackedPersistentQueryMetadata.getQueryId().toString()).isPresent()) {
+        log.debug("Registering query {} for validation", queryId);
+        try {
+          kafkaStreams.addNamedTopology(binpackedPersistentQueryMetadata.getTopologyCopy(this))
+              .all()
+              .get();
+        } catch (final Exception e) {
+          log.error(String.format("Validation for query %s failed due to:", queryId), e);
+          throw e;
+        }
+      } else {
+        log.info("Validation for a create or replace runtime");
+        try {
```

Comment:
I also made a small change here. When verifying a COR statement we can remove the old topology and try the new one instead of just skipping that part of validation

Reply:
Nice, only suggestion is a slight grammatical improvement to the logging, something like 
```
        log.info("Validating CREATE OR REPLACE for query {} in runtime {}", queryId, getApplicationId());
```